python/ml4ir/applications/ranking/t_test.py

```diff
@@ -1,5 +1,324 @@
-from scipy import stats
 import numpy as np
+import scipy
+import warnings
+from scipy import stats
+from scipy.optimize import brenth
+
+
+class StreamVariance:
+    # Wrapper class to store mean, variance in batches
+    count = 0
+    mean = 0
+    var = 0
+
+
+def power_ttest(d=None, n=None, power=None, alpha=0.05, contrast="two-samples", alternative="two-sided"):
+    """
+    Extracted from https://pingouin-stats.org/build/html/generated/pingouin.power_ttest.html
```

Comment:
(same comment as below)
@mohazahran Why not just add a pingouin dependency? Copy-pasting code means we don't get any of the new fixes/updates they make.

Make sure you request 3PP in parallel.

Reply:
I will do that. I believe till this happen we can stick to the extracted functions.

Reply:
I would prefer adding the dependency to this PR as opposed to checking in this source code from another repo if avoidable. Let's try to get the 3PP request going and see if it's delayed. Can decide next action based on that.

Also, the reason I ask for adding the dependency is I do not know which version of the library works well for our usecase. So it will be good to take it into consideration for the unittests and such.